### ISSUE TITLE
Improve upgrade process

### ISSUE THREAD

**Issue by c9s** [phpbrew/phpbrew#873]
the current upgrade command doesn't replace the existing install since the extraction only check if the "buildDir" exists, it doesn't overwrite the old files.

to upgrade the existing build, the items below might have to be handled:

1. update the build directory.
2. installed extensions needs to be re-compiled (for patch/minor version upgrade)
3. clean the object files before building.
4. config files shouldn't be overwritten.

**Comment by jasny** [comment]
@c9s What upgrade command?

**Comment by c9s** [comment]
@jasny for example, upgrade an existing PHP from 7.1.0 to 7.1.5 and 7.2.1 to 7.2.5

**Comment by jasny** [comment]
@c9s Yes, how would you do that? Without just installing a new version (7.1.5) and remove the old one (7.1.0).

**Comment by c9s** [comment]
For example, You can install 7.1 as “dev” into a specific location.

The upgrade should install new php into the same location. So that the
config files and extension files can be still used.

Arnold Daniels <notifications@github.com>於 2018年6月11日 週一，15:42寫道：
